core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -79,15 +79,23 @@ class TopicDeletionManager(controller: KafkaController,
   val controllerContext = controller.controllerContext
   val partitionStateMachine = controller.partitionStateMachine
   val replicaStateMachine = controller.replicaStateMachine
-  val topicsToBeDeleted: mutable.Set[String] = mutable.Set.empty[String] ++ initialTopicsToBeDeleted
-  val partitionsToBeDeleted: mutable.Set[TopicAndPartition] = topicsToBeDeleted.flatMap(controllerContext.partitionsForTopic)
   val deleteLock = new ReentrantLock()
   val topicsIneligibleForDeletion: mutable.Set[String] = mutable.Set.empty[String] ++
     (initialTopicsIneligibleForDeletion & initialTopicsToBeDeleted)
   val deleteTopicsCond = deleteLock.newCondition()
   val deleteTopicStateChanged: AtomicBoolean = new AtomicBoolean(false)
   var deleteTopicsThread: DeleteTopicsThread = null
   val isDeleteTopicEnabled = controller.config.deleteTopicEnable
+  val topicsToBeDeleted: mutable.Set[String] = mutable.Set.empty[String]
+  if(isDeleteTopicEnabled) {
+    topicsToBeDeleted ++= initialTopicsToBeDeleted
+  } else {
+    // if delete topic is disabled clean the topic entries under /admin/delete_topics
+    for(topic <- initialTopicsToBeDeleted) {
+      cleanZkStateForDeleteTopic(topic)
```

Comment:
If it is disabled then we should only remove the admin path right? It appears you are deleting the **entire state** for a topic (that isn't actually deleted).